CIP XXX Marlowe Oracle Protocol/v0.1.0-alpha.md

```diff
@@ -0,0 +1,252 @@
+* In the case that there is no published interface, an anchor that points to a file in IPFS containing a sufficient description to understand the particular data\_query, query\_result and resolution for this particular choice.
+
+The **data\_query** should contain a query to the data\_source that conforms to the API. 
+
+* In the case of the running example this will be a URL making a GET request to the endpoint for the service, defined according to the standard in the API.    
+* Instead of a query, the field may contain the anchor for an IPFS file containing the query.  
+* The query encoding can be also encoded using a custom format not suitable for human consumption which should be resolved and explained by the API. 
+
+The **result** should contain an anchor for an IPFS file that contains the result of the query, and a description of how an Integer is extracted from the result, which is called the *resolution* of the query. 
+
+* In the case of the running example, results are returned in JSON format, and the resolution will typically involve:
+  * Selecting a field or fields  
+  * Converting those fields to numbers  
+  * Aggregating those numbers into a single Integer.
+
+The **resolution** should contain an Integer, constructed according to the description in the **result**.
+
+### Data Availability
+
+Let's build up a minimal Marlowe contract request structure first. The goal is to make it useful in the data sharing context. We will then move to the security aspect of the data sharing and later on we will dive into the specifics of the datum and the contract state which should be used by the external on-chain data consumers.
+Some elements of this scheme require slight extensions to the current Marlowe validator (but not the Language) and its tooling. We will indicate those elements in the spec.
+
+#### Marlowe Oracle Request
+
+##### Minimal Marlowe Request
+
+As already discussed a minimal protocol request consists of two constructs a `Choice` and a `Payout`. This structure provides oracle harvester ability to easily detect requests on the chain together with a guarantee of the reward payment.
+
+  <img src="./diagrams/marlowe-request.svg" alt="Minimal Marlowe Request" style="width: 30%" />
+
+On its own this contract structure is not a reliable from the other data consumers point of view. Marlowe validator enforces removal of the contract thread UTxO together with the state and choices from the blockchain when a contract reaches its execution end. For example if we consider the above contract alone the data point will disappear from the chain together with transaction in which it is provided. We just reach the `Close` step together with the `Choice` and `Pay` in the same on-chain transaction - `Pay` and `Close` are evaluated "eagerly" after the `Choice` suspension point (they are not suspension points on themselves).
```

Comment:
This discussion suggests that a transaction will disappear from the chain: surely that's not right. Transactions are in the chain forever, surely?

Reply:
I added a section on that with diagram and reference it from this section.